gcp/bigquery/resources.py

```diff
@@ -15,7 +15,7 @@ def datasets():
             get_dataset(d["datasetReference"]["datasetId"], project_id)
             for d in datasets
         ]
-    return sum(results, [])
+    return results
```

Comment:
You're changing the return type here (from int to list) -- but I don't see any changes for where this function is invoked. Are we sure that code will know what to do?

Or maybe add typing to this function and see what mypy has to say?

Reply:
`sum(var, [])` collapses a list:

```
>>> x = [[1,2,3], ["foo", "bar"]]
>>> x
[[1, 2, 3], ['foo', 'bar']]
>>> sum(x, [])
[1, 2, 3, 'foo', 'bar']
```

This would be needed if we were doing `results.append([get_dataset(d["datasetReference"]["datasetId"], project_id) for d in datasets])`, but we are doing `results += [...]` so it isn't needed.

Reply:
Ah, so still returns a list. I didn't know about that use of sum, and thinking
```python
>>> sum([1, 2, 3])
6
```

TIL